Source/DataModels/Resources/ResourceNavMesh.cpp

```diff
@@ -0,0 +1,1032 @@
+	cfg.detailSampleMaxError = cellHeight * detailSampleMaxError;
+
+	rcVcopy(cfg.bmin, bmin);
+	rcVcopy(cfg.bmax, bmax);
+	int gw = 0, gh = 0;
+	rcCalcGridSize(bmin, bmax, cellSize, &gw, &gh);
+	const int ts = (int) tileSize;
+	const int tw = (gw + ts - 1) / ts;
+	const int th = (gh + ts - 1) / ts;
+
+	LOG_DEBUG("Building navigation:");
+	LOG_DEBUG(" - %d x %d cells", cfg.width, cfg.height);
+	LOG_DEBUG(" - %.1fK verts, %.1fK tris", verts.size() / 1000.0f, ntris / 1000.0f);
+
+	int tileBits = rcMin((int) dtIlog2(dtNextPow2(tw * th * EXPECTED_LAYERS_PER_TILE)), 14);
+	if (tileBits > 14)
+		tileBits = 14;
+	int polyBits = 22 - tileBits;
+	maxTiles = 1 << tileBits;
+	maxPolysPerTile = 1 << polyBits;
+
+	dtTileCacheParams tcparams;
+	memset(&tcparams, 0, sizeof(tcparams));
+	rcVcopy(tcparams.orig, bmin);
+	tcparams.cs = cellSize;
+	tcparams.ch = cellHeight;
+	tcparams.width = (int) tileSize;
+	tcparams.height = (int) tileSize;
+	tcparams.walkableHeight = agentHeight;
+	tcparams.walkableRadius = agentRadius;
+	tcparams.walkableClimb = agentMaxClimb;
+	tcparams.maxSimplificationError = edgeMaxError;
+	tcparams.maxTiles = tw * th * EXPECTED_LAYERS_PER_TILE;
+	tcparams.maxObstacles = 128;
+
+	dtFreeTileCache(tileCache);
+
+	tileCache = dtAllocTileCache();
+	if (!tileCache)
+	{
+		LOG_ERROR("buildTiledNavigation: Could not allocate tile cache.");
+		return false;
+	}
+	dtStatus status = tileCache->init(&tcparams, talloc, tcomp, tmproc);
+	if (dtStatusFailed(status))
+	{
+		LOG_ERROR("buildTiledNavigation: Could not init tile cache.");
+		return false;
+	}
+
+	dtFreeNavMesh(navMesh);
+
+	navMesh = dtAllocNavMesh();
+	if (!navMesh)
+	{
+		LOG_ERROR("buildTiledNavigation: Could not allocate navmesh.");
+		return false;
+	}
+
+	dtNavMeshParams params;
+	memset(&params, 0, sizeof(params));
+	rcVcopy(params.orig, bmin);
+	params.tileWidth = tileSize * cellSize;
+	params.tileHeight = tileSize * cellSize;
+	params.maxTiles = maxTiles;
+	params.maxPolys = maxPolysPerTile;
+
+	status = navMesh->init(&params);
+	if (dtStatusFailed(status))
+	{
+		LOG_ERROR("buildTiledNavigation: Could not init navmesh.");
+		return false;
+	}
+
+	navQuery = dtAllocNavMeshQuery();
+	status = navQuery->init(navMesh, 2048);
+	if (dtStatusFailed(status))
+	{
+		LOG_ERROR("buildTiledNavigation: Could not init Detour navmesh query");
+		return false;
+	}
+
+	rcChunkyTriMesh* chunkyMesh = new rcChunkyTriMesh;
+	if (!chunkyMesh)
+	{
+		LOG_ERROR("buildTiledNavigation: Out of memory 'm_chunkyMesh'.");
+		return false;
+	}
+	if (!rcCreateChunkyTriMesh(&verts[0], &tris[0], ntris, 256, chunkyMesh))
+	{
+		LOG_ERROR("buildTiledNavigation: Failed to build chunky mesh.");
+		return false;
+	}
+
+	for (int y = 0; y < th; ++y)
+	{
+		for (int x = 0; x < tw; ++x)
+		{
+			TileCacheData tiles[MAX_LAYERS];
+			memset(tiles, 0, sizeof(tiles));
+			int ntiles =
+				RasterizeTileLayers(&verts[0], verts.size(), ntris, ctx, chunkyMesh, x, y, cfg, tiles, MAX_LAYERS);
```

Comment:
```suggestion
				RasterizeTileLayers(&verts[0], static_cast<int>(verts.size()), ntris, ctx, chunkyMesh, x, y, cfg, tiles, MAX_LAYERS);
```
This raises a warning as well, try with this.